### PR TITLE
Add PlanNodeIds

### PR DESCRIPTION
**Pull request by xinzhu-cai**
Add plan node ids

### CODE REVIEW

**Comment by apavlo** [comment]
Please fix conflicts

File: src/include/planner/plannodes/abstract_scan_plan_node.h [hunk 1]
```diff
@@ -85,12 +85,14 @@ class AbstractScanPlanNode : public AbstractPlanNode {
...
    * @param database_oid database oid for scan
    * @param namespace_oid OID of the namespace
+   * @param plan_node_id plan node id
    */
   AbstractScanPlanNode(std::vector<std::unique_ptr<AbstractPlanNode>> &&children,
                        std::unique_ptr<OutputSchema> output_schema,
                        common::ManagedPointer<parser::AbstractExpression> predicate, bool is_for_update,
-                       catalog::db_oid_t database_oid, catalog::namespace_oid_t namespace_oid)
-      : AbstractPlanNode(std::move(children), std::move(output_schema)),
+                       catalog::db_oid_t database_oid, catalog::namespace_oid_t namespace_oid,
```
Comment (rohanaggarwal7997) [inline src/include/planner/plannodes/abstract_scan_plan_node.h:93]: Why is there no corresponding change in the builder?

File: src/optimizer/logical_operators.cpp [hunk 2]
```diff
@@ -40,23 +41,25 @@ BaseOperatorNodeContents *LogicalGet::Copy() const { return new LogicalGet(*this
...
   auto get = std::make_unique<LogicalGet>();
   get->database_oid_ = database_oid;
   get->namespace_oid_ = namespace_oid;
   get->table_oid_ = table_oid;
   get->predicates_ = std::move(predicates);
   get->table_alias_ = std::move(table_alias);
   get->is_for_update_ = is_for_update;
+  get->plan_node_id_ = plan_node_id;
   return Operator(std::move(get));
 }
```
Comment (rohanaggarwal7997) [inline src/optimizer/logical_operators.cpp:55]: Add plannodeid in == operator overloaded

File: src/optimizer/logical_operators.cpp [hunk 3]
```diff
@@ -91,13 +94,14 @@ bool LogicalGet::operator==(const BaseOperatorNodeContents &r) {
...
 Operator LogicalExternalFileGet::Make(parser::ExternalFileFormat format, std::string file_name, char delimiter,
-                                      char quote, char escape) {
+                                      char quote, char escape, plan_node_id_t plan_node_id) {
   auto get = std::make_unique<LogicalExternalFileGet>();
   get->format_ = format;
   get->file_name_ = std::move(file_name);
   get->delimiter_ = delimiter;
   get->quote_ = quote;
   get->escape_ = escape;
+  get->plan_node_id_ = plan_node_id;
```
Comment (rohanaggarwal7997) [inline src/optimizer/logical_operators.cpp:104]: Add in ==

File: src/optimizer/logical_operators.cpp [hunk 4]
```diff
@@ -154,9 +160,10 @@ common::hash_t LogicalQueryDerivedGet::Hash() const {
 //===--------------------------------------------------------------------===//
 BaseOperatorNodeContents *LogicalFilter::Copy() const { return new LogicalFilter(*this); }
 
-Operator LogicalFilter::Make(std::vector<AnnotatedExpression> &&predicates) {
+Operator LogicalFilter::Make(std::vector<AnnotatedExpression> &&predicates, plan_node_id_t plan_node_id) {
   auto op = std::make_unique<LogicalFilter>();
   op->predicates_ = std::move(predicates);
+  op->plan_node_id_ = plan_node_id;
```
Comment (rohanaggarwal7997) [inline src/optimizer/logical_operators.cpp:166]: Add in ==

File: src/optimizer/logical_operators.cpp [hunk 5]
```diff
@@ -189,9 +196,11 @@ common::hash_t LogicalFilter::Hash() const {
 //===--------------------------------------------------------------------===//
 BaseOperatorNodeContents *LogicalProjection::Copy() const { return new LogicalProjection(*this); }
 
-Operator LogicalProjection::Make(std::vector<common::ManagedPointer<parser::AbstractExpression>> &&expressions) {
+Operator LogicalProjection::Make(std::vector<common::ManagedPointer<parser::AbstractExpression>> &&expressions,
+                                 plan_node_id_t plan_node_id) {
   auto op = std::make_unique<LogicalProjection>();
   op->expressions_ = std::move(expressions);
+  op->plan_node_id_ = plan_node_id;
```
Comment (rohanaggarwal7997) [inline src/optimizer/logical_operators.cpp:203]: Add in == For all of them

**Comment by apavlo** [comment]
Closed to reduce Jenkins contention

**Pull request closed without merging**